dispatcher/utils.py

```diff
@@ -1,14 +1,25 @@
 import importlib
 from enum import Enum
-from typing import Callable, Optional
+from typing import Callable, Optional, Protocol, Type, Union, runtime_checkable
+
+
+@runtime_checkable
+class RunnableClass(Protocol):
+    def run(self, *args, **kwargs) -> None: ...
+
+
+MODULE_METHOD_DELIMITER = '.'
+
+
+DispatcherCallable = Union[Callable, Type[RunnableClass]]
 
 
 def resolve_callable(task: str) -> Optional[Callable]:
     """
     Transform a dotted notation task into an imported, callable function, e.g.,
 
-    awx.main.tasks.system.delete_inventory
-    awx.main.tasks.jobs.RunProjectUpdate
+    awx.main.tasks.system:delete_inventory
+    awx.main.tasks.jobs:RunProjectUpdate
```

Comment:
After change the DELIMITER this needs to be updated again:
```suggestion
    awx.main.tasks.system.delete_inventory
    awx.main.tasks.jobs.RunProjectUpdate
```